### PR TITLE
changes configuring restassuerd from static code to RequestSpecBuilder

### PR DESCRIPTION
**Pull request by lordofthejars**
#### Short description of what this resolves:

Instead of configuring RestAssured using static code, now users can inject `RequestSpecBuilder` in test with docker values initialized.

#### Changes proposed in this pull request:

- RestAssured static code is not called anymore
- `RequestSpecBuilder` is injected with docker values
-

**Fixes**: #530

### CODE REVIEW

File: docker/ftest-restassured/src/test/java/org/arquillian/cube/restassured/PingPongTest.java [hunk 1]
```diff
@@ -14,9 +16,18 @@
...
 
+    @ArquillianResource
+    RequestSpecBuilder requestSpecBuilder;
+
     @Test
     public void should_receive_ok_message() throws MalformedURLException, InterruptedException {
-        RestAssured.get().then().assertThat().body("status", equalTo("OK"));
+        RestAssured
+                .given()
+                .spec(requestSpecBuilder.build())
```
Comment (bartoszmajsak) [inline docker/ftest-restassured/src/test/java/org/arquillian/cube/restassured/PingPongTest.java]: Formatting might use a bit more indentation to mark GWT blocks.

**Comment by bartoszmajsak** [comment]
I have one general remark to this approach - it locks the user to the certain `rest-assured` version we use in Cube.

Wouldn't it be easier to just have all those configuration bits injectable to the test itself and let the user configure request spec as they wish?

**Comment by lordofthejars** [comment]
All these information are also provided by Docker Cube Core, so if we let user create their own `RequestSpecBuilder` then this extension does not have sense at all. Notice that this extension is used to avoid user having to repeat the same over and over again in all tests (I mean injecting fields and creating the class). If we decide to not creating the class `RequestSpecBuilder` class in extension, then this extension is merely a reporter extension.

**Comment by bartoszmajsak** [comment]
> All these information are also provided by Docker Cube Core

This is exactly my point - I'm not sure if there is a value in having it as an extension for the very reasons I mentioned in the previous comment (locking users with certain version of rest-assured)

**Comment by lordofthejars** [comment]
Well the idea is to avoid the repeat yourself in every test doing the same, building the `RequestSpecBuilder`. We can remove this PR and just leave restassured module as reporting features.

**Comment by lordofthejars** [comment]
What I would do is just squash this PR, and then open a new one with the removal. Just to make commits have the right change.

**Comment by bartoszmajsak** [comment]
Ok, if you feel like removing it makes sense, go for it.

**Pull request merged**